master/internal/resourcemanagers/resource_pool.go

```diff
@@ -38,38 +41,52 @@ type ResourcePool struct {
 
 // NewResourcePool initializes a new empty default resource provider.
 func NewResourcePool(
+	config *ResourcePoolConfig,
+	cert *tls.Certificate,
 	scheduler Scheduler,
 	fittingMethod SoftConstraint,
-	provisioner *actor.Ref,
-	provisionerSlotsPerInstance int,
 ) *ResourcePool {
 	d := &ResourcePool{
+		config: config,
+		cert:   cert,
+
 		scheduler:     scheduler,
 		fittingMethod: fittingMethod,
-		agents:        make(map[*actor.Ref]*agentState),
-		groups:        make(map[*actor.Ref]*group),
-
-		taskList: newTaskList(),
 
-		provisioner:      provisioner,
-		slotsPerInstance: provisionerSlotsPerInstance,
-		scalingInfo:      &sproto.ScalingInfo{},
+		agents:      make(map[*actor.Ref]*agentState),
+		taskList:    newTaskList(),
+		groups:      make(map[*actor.Ref]*group),
+		scalingInfo: &sproto.ScalingInfo{},
 
 		reschedule: false,
 	}
 	return d
 }
 
-func (d *ResourcePool) addTask(ctx *actor.Context, msg AllocateRequest) {
-	d.notifyOnStop(ctx, msg.TaskActor, ResourcesReleased{TaskActor: msg.TaskActor})
+func (rp *ResourcePool) setupProvisioner(ctx *actor.Context) error {
+	if rp.config.Provider == nil {
+		ctx.Log().Infof("disabling provisioner for resource pool: %s", rp.config.PoolName)
```

Comment:
non-blocking: would it be more accurate to say `not enabling provisioner ...`

Reply:
I don't really see the difference here except that `disabling` saves 3 chars.

Reply:
Disabling implies it was enabled at some point.